GP24-102-CSharp-Basic/NimA/Program.cs

```diff
@@ -0,0 +1,73 @@
+﻿int totalNumber = 24;
+int inputNumber;
+string shownMatches;
+Random random = new Random();
+int drawAI = random.Next(1, 4);
+
+
+Console.WriteLine("Welcome to NIM");
+Console.WriteLine("You can draw between 1 and 3 matches.");
+DisplayMatches();
+
+while (totalNumber > 0)
+{
+    PlayerTurn();
+    AIturn();
+}
+GameResult();
+
+void DisplayMatches()
+{
+    shownMatches = new string('|', totalNumber);
+    Console.WriteLine(shownMatches + $" ({totalNumber})");
+}
+
+void PlayerTurn()
+{
+    Console.WriteLine();
+    Console.WriteLine("How many matches do you want to draw?");
+    inputNumber = Convert.ToInt32(Console.ReadLine());
+    
+    if (inputNumber >0 && inputNumber <=3)
+    {
+        totalNumber = totalNumber - inputNumber;
+        DisplayMatches();
+    }
+    else
+    {
+        Console.WriteLine("This is not a valid number for the game. Try again.");
+        DisplayMatches();
+    }
+}
+
+void AIturn()
+{
+    Console.WriteLine();
+    Console.WriteLine("Now it is the AI's turn to draw");
+    Console.WriteLine($"The AI drew {drawAI} match(es)");
+    totalNumber = totalNumber - drawAI;
+    DisplayMatches();
```

Comment:
The problem is, if you don't recalculate `drawAI` here, then the AI will draw the same amount of matches every turn (it only gets randomized once on app start).

START:
aiDraw = (random) 2;

void AITurn(){
   // will now always draw 2 matches, because you never assign to aiDraw again
}